### PR TITLE
Upgrade Dependency Ranges for 1.1

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Upgrade Dependency Ranges for 1.1
- Update pandas to 2.2.x has been moved to a future PR.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: core/src/autogluon/core/_setup_utils.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 DEPENDENT_PACKAGES = {
     "boto3": ">=1.10,<2",  # <2 because unlikely to introduce breaking changes in minor releases. >=1.10 because 1.10 is 3 years old, no need to support older
     "numpy": ">=1.21,<1.29",  # "<{N+3}" upper cap, where N is the latest released minor version, assuming no warnings using N
-    "pandas": ">=2.0.0,<2.2.0",  # "<{N+1}" upper cap
+    "pandas": ">=2.0.0,<2.3.0",  # "<{N+1}" upper cap
```
Comment (shchur) [inline core/src/autogluon/core/_setup_utils.py:21]: Pandas 2.2 introduced a breaking change to how time frequency aliases are represented as strings (https://pandas.pydata.org/docs/whatsnew/v2.2.0.html#other-deprecations). For example, `pandas.tseries.frequencies.to_offset("1min")` returns 
- `min` in pandas >=2.2
- `T` in pandas <2.2
I will need to change our logic for detecting seasonality to take this into account.

Comment (Innixma) [reply core/src/autogluon/core/_setup_utils.py]: Moving pandas upgrade to a separate PR to isolate the change

**Review by suzhoum — APPROVED** [review]
LGTM!

**Review by prateekdesai04 — COMMENTED** [review]
LGTM

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |
| vowpalwabbit==9.9.0 | vowpalwabbit==9.8.0 |
| skl2onnx==1.16.0 | skl2onnx==1.15.0 |
| vowpalwabbit==9.9.0 | vowpalwabbit==9.8.0 |
| skl2onnx==1.16.0 | skl2onnx==1.15.0 |

**Pull request merged**